src/js/dgram.js

```diff
@@ -132,7 +139,7 @@ Socket.prototype.bind = function(port /*, address, callback*/) {
 
     self._handle._reuseAddr = self._reuseAddr;
 
-    var err = self._handle.bind(ip, port || 0);
+    var err = self._handle.bind(ip, port | 0);
```

Comment:
I made it stronger to get error early. E.g. if port is an empty `{}` object, then `{} || 0` returns with `{}` and we get an error later in the native part. However the `| 0` forces an int32 conversion, which is enough for a port number. Also `new Number(13) | 0` will give back 13, rather than an object, which simplifies conversions later as well (e.g. native functions complains on type mismatch).